byway/src/main/webapp/track-interests.js

```diff
@@ -92,6 +93,22 @@ function createButtonForPlace(place) {
   return button;
 }
 
+/* Hide or display list of interests to match user input as a zero-indexed substring. */
+function filterInterests() {
+  const searchField = document.getElementById("search-interest");
+  const searchValue = searchField.value.toLowerCase();
```

Comment:
You may not want to filter anything out if the search value is empty.

Reply:
It would be a nice check to be done initially, but consider this: a user types in "a," it filters for the interests starting with "a," and then the user deletes the "a" - say he meant to type in "b" instead. Since this function is called per `keyup` event, `filterInterests` would see no input and return to avoid the for loop. It would skip the filtering, which is in charge of either hiding or showing elements. It would be stuck displaying only interests starting with "a" despite there being no input, which should show all elements.